app/models/reservation.rb

```diff
@@ -1,7 +1,7 @@
 # frozen_string_literal: true
 
 class Reservation < ApplicationRecord
-  belongs_to :screening, :user
-  has_many :tickets
-  validates_associated :tickets
+  belongs_to :screening
+  belongs_to :user, optional: true
+  has_and_belongs_to_many :seats
```

Comment:
We shouldn't use `has_and_belongs_to_many` relationship. Read about [this](https://flatironschool.com/blog/why-you-dont-need-has-and-belongs-to-many)

Reply:
Thanks, I change this relationship to `has many through`.